hail/python/hailtop/batch/batch.py

```diff
@@ -149,6 +151,64 @@ def __init__(self,
 
         self._cancel_after_n_failures = cancel_after_n_failures
 
+        self._python_function_defs: Dict[int, Callable] = {}
+        self._python_function_files: Dict[int, Optional[_resource.InputResourceFile]] = {}
+
+    def _get_python_function(self, function_id: int) -> Callable:
+        function = self._python_function_defs.get(function_id, None)
+        if function is None:
+            raise ValueError(
+                f"No Python function registered with id: {function_id}."
+            )
+        return function
+
+    def _get_python_function_file(self, function_id: int) -> _resource.InputResourceFile:
+        try:
+            file = self._python_function_files[function_id]
+            if file is None:
+                raise ValueError(
+                    f"No file found for Python function registered with id: {function_id}."
+                )
+            return file
+        except KeyError as error:
+            raise ValueError(
+                f"No Python function registered with id: {function_id}."
+            ) from error
```

Comment:
Both this method and the field it protects are private and this PR creates an invariant where we'll only make valid accesses into this `_python_function_files` dictionary (and if we didn't, the underlying `KeyError` would be informative enough). So I think we don't need this function and can just access the dictionary directly where this is used.

Reply:
Updated!